packages/reference/src/resources/useResourceLinkActions.ts

```diff
@@ -25,39 +32,21 @@ export function useResourceLinkActions({
   );
   const multiple = field.type === 'Array';
 
-  const toLinkItem = useMemo(() => {
-    function toUrn(entry: EntryProps<KeyValueMap>) {
-      return `crn:${apiUrl}:::content:spaces/${entry.sys.space.sys.id}/entries/${entry.sys.id}`;
-    }
-
-    return (entry: EntryProps<KeyValueMap>) => {
-      return {
-        sys: {
-          type: 'ResourceLink',
-          linkType: 'Contentful:Entry',
-          urn: toUrn(entry),
-        },
-      };
-    };
-  }, [apiUrl]);
-
   const onLinkedExisting = useMemo(() => {
-    if (multiple) {
-      return (entries: EntryProps<KeyValueMap>[]) => {
-        const linkItems = entries.map(toLinkItem);
+    return (entries: EntryProps<KeyValueMap>[]) => {
```

Comment:
Was there a specific itch you wanted to scratch by changing it to a single function? Doesn't feel more readable to me.

Reply:
We did this with Luca while pair programming. I'm ok with reverting it. What do you think? @Bikappa

Reply:
I avoids the duplication of the code that is now around the if/else block. Nothing massive oc, but it stops 2 chunks of code that do the same thing from growing

Reply:
Perhaps we can bring back readability with (pseudocode):
```
...
const updatedValue = giveMeTheValue(multiple, prevValue, entriesWithResourceName)
...
```

Reply:
Don't want to bikeshed, but it's worth considering keeping 2 lines of duplicated code for the sake of readibility.

Reply:
Ok I have extracted it to a function. I think it's readable now and also there's no code duplication. We can all agree that extraction is a good practice as well :) I'll merge this tomorrow